src/options/output.rs

```diff
@@ -0,0 +1,203 @@
+                        RoverOutput::FetchResponse(fetch_response) => {
+                            match fetch_response.sdl.r#type {
+                                SdlType::Graph | SdlType::Subgraph { .. } => "SDL",
+                                SdlType::Supergraph => "Supergraph SDL",
+                            }
+                        }
+                        RoverOutput::CoreSchema(_) => "CoreSchema",
+                        RoverOutput::CompositionResult(_) => "CoreSchema",
+                        RoverOutput::TemplateUseSuccess { .. } => "Project generated",
+                        RoverOutput::CheckResponse(_) => "Check Result",
+                        RoverOutput::AsyncCheckResponse(_) => "Check Started",
+                        RoverOutput::Profiles(_) => "Profiles",
+                        RoverOutput::Introspection(_) => "Introspection Response",
+                        RoverOutput::ReadmeFetchResponse { .. } => "Readme",
+                        RoverOutput::GraphPublishResponse { .. } => "Schema Hash",
+                        _ => return Ok(()),
+                    };
+                    if let RoverOutput::GraphPublishResponse { .. } = self {
+                        self.print_one_line_descriptor(descriptor)?;
+                    } else {
+                        self.print_descriptor(descriptor)?;
+                    }
+
+                    stdoutln!("{}", &result)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl RoverOutputTrait for RoverError {
+    fn write_or_print(self, format_type: FormatType, _: FinalOutputType) -> RoverResult<()> {
+        match format_type {
+            FormatType::Plain => self.print(),
+            FormatType::Json => JsonOutput::from(self).print(),
+        }?;
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser, Serialize)]
+pub struct OutputStrategy {
+    /// Specify Rover's format type
+    #[arg(long = "format", global = true)]
+    format_type: Option<FormatType>,
+
+    /// Specify a file to write Rover's output to
+    #[arg(long = "output", global = true)]
+    output_type: Option<OutputType>,
+}
+
+impl OutputStrategy {
+    pub fn validate_options(&self) {
+        match (&self.format_type, &self.output_type) {
+            (Some(_), Some(OutputType::LegacyOutputType(_))) => {
+                let mut cmd = Rover::command();
+                cmd.error(
+                    ClapErrorKind::ArgumentConflict,
+                    "The argument '--output' cannot be used with '--format' when '--output' is not a file",
+                )
+                .exit();
+            }
+            (None, Some(OutputType::LegacyOutputType(_))) => {
+                let warn_prefix = Style::WarningPrefix.paint("WARN:");
+                eprintln!("{} The argument '--output' will soon be deprecated. Please use the '--format' argument to specify the output type.", warn_prefix);
+            }
+            _ => (),
```

Comment:
let's add a comment here explaining why we're skipping this section
```suggestion
            // there are default options, so if nothing is passed, print no errors or warnings
            _ => (),
```